### DISCUSSION TITLE
How to display gray8 and rgb24  images?

### DISCUSSION THREAD

**Discussion by qfmee** [AvaloniaUI/Avalonia#11711] [General]
new Bitmap , Avalonia.Platform.PixelFormat only have Bgra8888 、Rgba8888、Rgb565 !How to display gray8 and rgb24  images?
Especially during camera cycle collection and display, if gray8 is converted to rgba8888, it will be a waste of computer resources，Is there a more efficient way?

**Comment by maxkatz6** [comment]
Avalonia 11 has more possible pixel formats https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Platform/PixelFormat.cs#L65-L79

**Reply by qfmee** [reply]
I have another question: Bitmap Bitmap=new Bitmap (fileName); The image source is a gray image. How to read the gray format instead of the rgba8888 format? Thank you

**Reply by maxkatz6** [reply]
At the moment, you only can decode Bitmap without expecting any specific format. As it depends on the platform backend, and Skia specifically. Skia doesn't support gray image pixel format with avalonia.
So instead you can only transcode it from rgba8888  to gray after decoding. Possibly using WriteableBitmap and constructors accepting pointer to the data and expected pixel format - it will convert data array to the proper view.

**Comment by qfmee** [comment]
Thank you! Avalonia.Platform.PixelFormats is different from Avalonia.Platform.PixelFormat
